### ISSUE TITLE
Add filter/conversion function that can be applied to editable cells

### ISSUE THREAD

**Issue by bontibon** [AllenFang/react-bootstrap-table#631]
**Version:** 2.4.2

`beforeSaveCell` or `onCellEdit` should be allowed to return a new value that is then used for `newVal`.

**Comment by AllenFang** [comment]
@bontibon, could you please describe it more detail?

**Comment by bontibon** [comment]
1. I have an editable text column
2. When the user clicks the column, the user can edit the value in an input box
3. When the input box blurs
  - `beforeSaveCell` is called
  - `onCellEdit` is called (**I want to return value to be used as the `cellValue`**)
  - `afterSaveCell` is called
4. the value is saved back to the data object

The change would happen around [here](https://github.com/AllenFang/react-bootstrap-table/blob/2e6144c9994ce38d03fff58d50c04f04f343647d/src/BootstrapTable.js#L503):

Replace
```javascript
    if (onCellEdit) {
      onCellEdit(this.state.data[rowIndex], fieldName, newVal);
    }
```

With
```javascript
    if (onCellEdit) {
      newVal = onCellEdit(this.state.data[rowIndex], fieldName, newVal);
    }
```

**Comment by AllenFang** [comment]
@bontibon, sorry for lately reply. I'll think about your requirement, it seems to be ok. :)

**Comment by AllenFang** [comment]
Released on ```v2.4.3```
